zerver/views/user_settings.py

```diff
@@ -34,8 +34,11 @@
     EmailChangeStatus
 from confirmation.models import EmailChangeConfirmation
 
-def confirm_email_change(request, confirmation_key):
-    # type: (HttpRequest, str) -> HttpResponse
+def confirm_email_change(request, user_profile, confirmation_key):
+    # type: (HttpRequest, UserProfile, str) -> HttpResponse
+    if user_profile.realm.email_changes_disabled:
+        raise JsonableError(_("Email change disabled"))
+
```

Comment:
I think it would be reasonable to add the `@zulip_login_required` decorator to this function.